### ISSUE TITLE
Defining internal-regex and import/external-module-folders together yields unexpected errors

### ISSUE THREAD

**Issue by Slapbox** [benmosher/eslint-plugin-import#2139]
In our monorepo we store our internal packages under `./packages`

So we tried a config like this:

```js
rules: {
  'import/order': 'warn',
},
settings: {
  'import/internal-regex': '^internal-', //  This makes importing monorepo packages get treated as though they were 'relative-parent-imports'
  'import/external-module-folders': ['node_modules', 'packages'],
}
```

But that makes a non-relative import throw this unexpected error

```js
import { foo } from 'internal-utils';
                              ^--- Relative imports from parent directories are not allowed.
```

**Comment by Slapbox** [comment]
We were able to workaround this with the following configuration in the meantime:


```js
rules: {
  'import/order': [
    'error',
    {
      pathGroups: [
        {
          pattern: 'internal-*',
          group: 'external',
          position: 'after',
        },
      ],
      pathGroupsExcludedImportTypes: [],
    },
  ],
}, 
settings: {
  // 'import/internal-regex': '^internal-', // Disable this line
  'import/external-module-folders': ['node_modules', 'packages'],
}
```

**Comment by soryy708** [comment]
Interesting interaction between the rules.

Would be nice if anybody contributes a [test case](https://github.com/import-js/eslint-plugin-import/tree/668d493dbb456415c3aa12bcbde7917d475393c5/tests)
